ui/apps/akasha/src/components/feed-page/feed-page.tsx

```diff
@@ -41,18 +41,25 @@ const FeedPage: React.FC<FeedPageProps & RootComponentProps> = props => {
 
   const [analyticsActions] = useAnalytics();
 
+  const controller = new AbortController();
+
   //get the post id for repost from the search param
   const [postId, setPostId] = React.useState(new URLSearchParams(location.search).get('repost'));
 
   React.useEffect(() => {
-    /*The popstate listener is required for reposts happening from the feed page */
+    /*The single-spa:before-routing-event listener is required for reposts happening from the feed page */
     const popStateHandler = () => {
       setPostId(new URLSearchParams(location.search).get('repost'));
     };
-    window.addEventListener('popstate', popStateHandler);
+    window.addEventListener('single-spa:before-routing-event', popStateHandler, {
+      signal: controller.signal,
+    });
     return () => {
-      window.removeEventListener('popstate', popStateHandler);
+      window.removeEventListener('single-spa:before-routing-event', () => {
+        controller.abort();
```

Comment:
you don't need to call removeEventListener, also create the abort controller inside the useEffect